src/main/java/com/teragrep/pth_06/ArchiveMicroStreamReader.java

```diff
@@ -135,25 +163,70 @@ public final class ArchiveMicroStreamReader implements MicroBatchStream {
      */
     @Override
     public Offset initialOffset() {
+        // After rebase is complete: Refactor the DatasourceOffset and SerializedDatasourceOffset if the 8x else-if statements are too much.
```

Comment:
Indeed this if-else tree is getting too long, should be refactored

Reply:
Solved in commit 03552980d1289068474c778d099b2d0dc979c93f